### PR TITLE
add write api in FileUtils

### PR DESCRIPTION
**Pull request by jianglong0156**
add write api in FileUtils

```
bool writeStringToFile(std::string dataStr, const std::string& fullPath);
bool writeDataToFile(Data retData, const std::string& fullPath);
bool writeValueMapToFile(ValueMap& dict, const std::string& fullPath);
bool writeValueVectorToFile(ValueVector vecData, const std::string& fullPath);
```

and
formate the FileUtils code

### CODE REVIEW

**Comment by jianglong0156** [comment]
relate issue:https://github.com/cocos2d/cocos2d-x/issues/9621

**Comment by jianglong0156** [comment]
@WenhaiLin  Please review it

**Comment by jianglong0156** [comment]
[ci rebuild]

**Comment by jianglong0156** [comment]
@dualface  hi, I add the new api in FileUtils, it's need to be added in luaTest

File: cocos/platform/CCFileUtils.cpp [hunk 1]
```diff
@@ -394,17 +399,58 @@ bool FileUtils::writeToFile(ValueMap& dict, const std::string &fullPath)
...
+
+    bool ret = tinyxml2::XML_SUCCESS == doc->SaveFile(getSuitableFOpen(fullPath).c_str());
+
+    delete doc;
+    return ret;
+}
+
+bool FileUtils::writeValueVectorToFile(ValueVector vecData, const std::string& fullPath)
+{
+    //CCLOG("tinyxml2 Dictionary %d writeToFile %s", dict->_ID, fullPath.c_str());
```
Comment (WenhaiLin) [inline cocos/platform/CCFileUtils.cpp]: Please remove this line.

**Comment by jianglong0156** [comment]
@WenhaiLin  done

File: cocos/platform/CCFileUtils.cpp [hunk 2]
```diff
@@ -394,17 +398,57 @@ bool FileUtils::writeToFile(ValueMap& dict, const std::string &fullPath)
...
     bool ret = tinyxml2::XML_SUCCESS == doc->SaveFile(getSuitableFOpen(fullPath).c_str());
-    
+
+    delete doc;
+    return ret;
+}
+
+bool FileUtils::writeValueVectorToFile(ValueVector vecData, const std::string& fullPath)
+{
+    tinyxml2::XMLDocument *doc = new tinyxml2::XMLDocument();
```
Comment (WenhaiLin) [inline cocos/platform/CCFileUtils.cpp]: Please add `std::nothrow` to the `new` statements

**Comment by jianglong0156** [comment]
@WenhaiLin done

**Comment by WenhaiLin** [comment]
@jianglong0156 Review done. @pandamicro It's OK for merge.

**Pull request merged**